src/inputs/ToggleButton.test.tsx

```diff
@@ -0,0 +1,104 @@
+import { click, render, wait } from "src/utils/rtl";
+import { ToggleButton } from "./ToggleButton";
+
+describe("ToggleButton", () => {
+  it("renders", async () => {
+    // Given a ToggleButton
+    const r = await render(
+      <ToggleButton
+        label={"Toggle button"}
+        onClick={() => {}}
+      />
+    );
+    // Then it renders
+    expect(r.toggleButton().textContent).toBe("Toggle button");
+  });
+
+  it("renders selected state", async () => {
+    // Given a selected ToggleButton
+    const r = await render(
+      <ToggleButton
+        label={"Toggle button"}
+        selected
+        onClick={() => {}}
+      />
+    );
+    // Then it's checked
+    expect(r.toggleButton_value()).toBeChecked();
+
+    //Given a non selected ToggleButton
+    await r.rerender(
+      <ToggleButton
+        label={"Toggle button"}
+        onClick={() => {}}
+      />
+    );
+    // Then it isn't checked
+    expect(r.toggleButton_value()).not.toBeChecked();
```

Comment:
**suggestion:** What do you think about removing this extra render, and instead validating `not.toBeChecked` in the `it("renders")` test?